contracts/tokens/RewardEthToken.sol

```diff
@@ -152,26 +177,28 @@ contract RewardEthToken is IRewardEthToken, OwnablePausableUpgradeable, ERC20Per
 
         // calculate reward per token used for account reward calculation
         uint256 maintainerReward = periodRewards.mul(maintainerFee).div(10000);
-        rewardPerToken = rewardPerToken.add(periodRewards.sub(maintainerReward).mul(1e18).div(stakedEthToken.totalDeposits()));
+        uint256 newRewardPerToken = uint256(rewardPerToken).add(periodRewards.sub(maintainerReward).mul(1e18).div(stakedEthToken.totalDeposits()));
 
         // update maintainer's reward
-        checkpoints[maintainer] = Checkpoint(
-            rewardPerToken,
-            balanceOf(maintainer).add(maintainerReward)
+        Checkpoint memory checkpoint = Checkpoint(
+            balanceOf(maintainer).add(maintainerReward).toUint128(),
+            newRewardPerToken.toUint128()
         );
+        checkpoints[maintainer] = checkpoint;
 
+        totalRewards = newTotalRewards.toUint128();
+        rewardPerToken = newRewardPerToken.toUint128();
```

Comment:
The Solidity optimizer fails to pack the above two assignments into a single storage update, but works for the following:
```suggestion
        (totalRewards, rewardPerToken) = (newTotalRewards.toUint128(), newRewardPerToken.toUint128());
```

(Compare the gas cost between the two versions.)

Reply:
I've compared the gas cost and your suggested assignment saves around `1000` gas.